kyaml/setters2/list.go

```diff
@@ -192,3 +193,44 @@ func (l *List) count(path, name string) (int, error) {
 
 	return s.Count, err
 }
+
+// CheckRequiredSettersSet iterates through all the setter definitions in openAPI
+// file and returns error if any of the setter has required filed true and isSet false
+func CheckRequiredSettersSet(openAPIPath string) error {
+	oa, err := ioutil.ReadFile(openAPIPath)
+	if err != nil {
+		return nil
+	}
+
+	openAPI, err := yaml.Parse(string(oa))
+	if err != nil {
+		return nil
+	}
+
+	definitions, err := openAPI.Pipe(yaml.Lookup("openAPI", "definitions"))
+	if err != nil {
+		return nil
+	}
+
+	keys, err := definitions.Fields()
+	if err != nil {
+		return nil
+	}
+
+	for _, key := range keys {
+		required, err := definitions.Pipe(yaml.Lookup(key, "x-k8s-cli", "setter", "required"))
+		if required == nil || err != nil {
+			continue
+		}
+		requiredVal := required.Document().Value
+		println(requiredVal)
+
+		isSet, err := definitions.Pipe(yaml.Lookup(key, "x-k8s-cli", "setter", "isSet"))
```

Comment:
maybe just test err != nil by itself?

then check the requiredVal stuff?

Reply:
Updated the PR with new implementation.